include/eve/arch/cpu/as_register.hpp

```diff
@@ -61,91 +62,54 @@ namespace eve
 
   namespace detail
   {
-    template<typename Type, typename Cardinal> struct blob
+    template<typename Type, typename Cardinal>
+    struct  blob
     {
+      using is_product_type             = void;
       using cardinal_t                  = expected_cardinal_t<Type>;
-      static constexpr auto replication = Cardinal::value/cardinal_t::value;
-
       using value_type                  = as_wide_t<Type, cardinal_t>;
       using subvalue_type               = as_wide_t<Type, typename Cardinal::split_type>;
+      static constexpr auto replication = Cardinal::value/expected_cardinal_v<Type>;
 
-      std::array<subvalue_type,2>       segments;
+      std::array<value_type, replication> storage;
 
-      template<std::size_t I> EVE_FORCEINLINE decltype(auto) get() noexcept
+      EVE_FORCEINLINE void combine(subvalue_type const& l, subvalue_type const& h)
```

Comment:
I don't like this name.
`combine` should no sassign members to self.
 Let's talk

Reply:
As said before, this just could be a constructor.

Reply:
constructor or `assign` methods are good. If you don't like constructor, it's ok.

Also we have the constructor from halves and `eve::combine`. Why not put this logic directly there?

Reply:
I choose `assign_parts`